app/services/course/assessment/submission/update_service.rb

```diff
@@ -102,6 +108,23 @@ def edit_submission_path
     edit_course_assessment_submission_path(current_course, @assessment, @submission)
   end
 
+  # Find the questions for this submission without submission_questions.
+  # Build and save! new submission_questions.
+  #
+  # @return[Boolean] If new submission_questions were created.
+  def build_missing_submission_questions
```

Comment:
I think the method should be named `create_missing_xxx`, because you actually created them other than build.